### ISSUE TITLE
Search Page - Disable search filtering of Education Requirement

### ISSUE THREAD

**Issue by brindasasi** [GCTC-NTGC/gc-digital-talent#5976]
# Description
On the search page, it is possible to filter out users without a post-secondary education.
![image.png](https://images.zenhubusercontent.com/5d9ded8fd25901000101f3f0/b438c5f2-e2af-403e-bb76-b7fb4ccd505f)
In the backend, this adds the `hasDipoma=true` filter to the request.

However, while we previously allowed DCM to set the `hasDiploma` field when manually creating Pool Candidates, it doesn't appear anywhere in the profile! This means that any user who signs up and completes their own profile will automatically be excluded from any requests which require a diploma.

We have decided to disable the search filtering so as to not exclude anybody until the new education experience logic is complete.  However, the choice should still get communicated to DCM so that can consider in their work.

# Acceptance Criteria
- [ ] Search page selection of "required diploma" does NOT filter out any results
- [ ] Update `countPoolCandidates` and `countUsers` queries (approximate names for queries, might be slightly different) to not include that filter in counts
- [ ] Search page selection of "required diploma" DOES get saved in request that DCM receives
- [ ] Admin site searches including "required diploma" does NOT filter out any results
- [ ] has_diploma field in schema marked as deprecated

*Cloned from #4970 as per [this](https://github.com/GCTC-NTGC/gc-digital-talent/issues/4970#issuecomment-1331338130)

**Comment by brindasasi** [comment]
@Jerryescandon I'm unsure this is still valid .. but just creating this placeholder issue for your review .

**Comment by vd1992** [comment]
What is the state of this issue now that we've got a separate education field now attached to pool candidates? 
`PoolCandidate.educationRequirementOption` can be `APPLIED_WORK` or `EDUCATION`

Those two options basically map to the radio group. So is it still disable education filtering, or replace what field education filtering acts on?

**Comment by mnigh** [comment]
@gobyrne any update on this issue?

**Comment by gobyrne** [comment]
I think we need to disable the search filtering until we have Record of Decision. The problem with the current setup is that there is no review of the education requirement and the self-reported version we have now is not reliable enough.

ROD will have a step where Pool Operators review the education requirement to ensure it's accurate. So let's wait until then.

TLDR: Disable it
